tests/unit/commands/test_completions.py

```diff
@@ -0,0 +1,143 @@
+from argparse import Action
+from collections import namedtuple
+from typing import Tuple
+
+import pytest
+
+from borgmatic.commands.completion import (
+    dedent_strip_as_tuple,
+    exact_options_completion,
+    has_choice_options,
+    has_exact_options,
+    has_file_options,
+    has_unknown_required_param_options,
+)
+
+OptionType = namedtuple('OptionType', ['file', 'choice', 'unknown_required'])
```

Comment:
I think this is a prime candidate for a `flexmock()` instance. It _should_ be a drop-in replacement. Although maybe the one thing you can't do easily with a flexmock instance is the `if True in ...` you have below.

Reply:
Why would I use a mock for this? My goal was to create a way to reuse the test cases between the different functions when some inputs would match for multiple functions--I used this tuple to replace having independent lists for each function that were ``(Action, Boolean)``
Is there a reason to mock it out?

Reply:
Nope, I made this comment without realizing that these tuples aren't actually passed in anywhere. Given that they're just for test data structuring, I think they're fine as-is!